gc/base/SparseAddressOrderedFixedSizeDataPool.hpp

```diff
@@ -0,0 +1,277 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *
+******************************************************************************/
+
+#if !defined(SparseAddressOrderedFixedSizeDataPool_HPP_)
+#define SparseAddressOrderedFixedSizeDataPool_HPP_
+
+#include "omrpool.h"
+#include "BaseVirtual.hpp"
+#include "EnvironmentBase.hpp"
+#include "GCExtensionsBase.hpp"
+#include "SparseHeapLinkedFreeHeader.hpp"
+
+/**
+ *
+ * @ingroup GC_Base_Core
+ */
+
+class MM_SparseDataTableEntry
+{
+/*
+ * Data members
+ */
+private:
+public:
+	void *_dataPtr { NULL }; /**< Object data pointer related to proxy object */
+	void *_proxyObjPtr { NULL }; /**< Pointer to proxy object that is residing in-heap */
+	uintptr_t _size { 0 }; /**< Total size of the data pointed to by dataPtr */
```

Comment:
we don't use this style of intialization - move it to constructor(s)

Reply:
Done 👍